src/networkWrapper/wrappers/socketio.js

```diff
@@ -12,11 +22,75 @@ function SocketIO(host, port, ssl) {
    * @param {int} reconnectionDelay
    */
   this.connect = function (autoReconnect, reconnectionDelay) {
+    var self = this;
+
     this.socket = window.io((this.ssl ? 'https://' : 'http://') + this.host + ':' + this.port, {
       reconnection: autoReconnect,
       reconnectionDelay: reconnectionDelay,
       forceNew: true
     });
+
+    this.socket.on('connect', function() {
+      if (self.wasConnected) {
+        self.handlers.reconnect.forEach(function(handler) {
+          handler();
+        });
+      }
+      else {
+        self.handlers.connect.forEach(function(handler) {
+          handler();
+        });
+      }
+
+      self.wasConnected = true;
+      self.stopRetryingToConnect = false;
+    });
+
+    this.socket.on('connect_error', function(error) {
+      self.handlers.connectError.forEach(function(handler) {
+        handler(error);
+      });
+
+      if (autoReconnect && !self.retrying && !self.stopRetryingToConnect) {
+        self.retrying = true;
+        setTimeout(function () {
+          self.retrying = false;
+          self.connect(autoReconnect, reconnectionDelay);
+        }, reconnectionDelay);
+      }
+    });
+
+    this.socket.on('disconnect', function() {
+      if (!self.forceDisconnect) {
+        self.handlers.disconnect.forEach(function(handler) {
+          handler();
+        });
+      }
+
+      self.forceDisconnect = false;
+    });
+
+    this.socket.on('kuzzle_proxy_disconnection', function(error) {
+      self.forceDisconnect = true;
+
+      self.handlers.connectError.forEach(function(handler) {
+        handler(error);
+      });
+
+      if (autoReconnect && !self.retrying) {
+        self.retrying = true;
+        setTimeout(function () {
+          self.retrying = false;
+          self.connect(autoReconnect, reconnectionDelay);
+        }, reconnectionDelay);
+      }
```

Comment:
I don't know why but reconnection from socketio does not work... (love socketio <3)